src/compiler/checker.ts

```diff
@@ -9273,7 +9296,9 @@ module ts {
             var staticType = <ObjectType>getTypeOfSymbol(symbol);
             var baseTypeNode = getClassBaseTypeNode(node);
             if (baseTypeNode) {
-                emitExtends = emitExtends || !isInAmbientContext(node);
+                if (languageVersion < ScriptTarget.ES6) {

```

Comment:
I think i'd prefer the language check in the emitter.  THe checker is essentially saying "yes, something extended something else".  The emitter decides "ok, i need to emit __extends or not depending on the target version."